reframe/frontend/cli.py

```diff
@@ -1111,7 +1110,11 @@ def _case_failed(t):
             list_checks(testcases, printer)
             printer.info('[Generate CI]')
             with open(options.ci_generate, 'wt') as fp:
-                ci.emit_pipeline(fp, testcases)
+                child_pipeline_opts = []
+                if options.mode:
+                    child_pipeline_opts.append(f'--mode={options.mode}')
+
+                ci.emit_pipeline(fp, testcases, child_pipeline_opts)
```

Comment:
I was thinking of passing directly `options.mode` here. Why did yo prefer to do it this way?

Reply:
Just to check that `--mode` was passed and avoid passing `--mode=None` to the pipeline.

Reply:
Eventually, I think it makes sense, because otherwise you would have to reparse the mode options.